### PR TITLE
fix: Replace broken `fetchDeployComponentsMetadata` method

### PR DESCRIPTION
**Pull request by joe-yeager**
## Description and Context

Companion PR for https://github.com/HubSpot/hubspot-cli/pull/1131/files

- Replace a broken method with the new API
- Test the `api/projects.ts` since we are updating it.

## Who to Notify

@kemmerle @brandenrodgers @camden11

### CODE REVIEW

File: api/projects.ts [hunk 1]
```diff
@@ -64,7 +65,7 @@ export async function uploadProject(
 
   return http.post(accountId, {
     url: `${PROJECTS_API_PATH}/upload/${encodeURIComponent(projectName)}`,
-    timeout: 60000,
+    timeout: 60_000,
```
Comment (joe-yeager) [inline api/projects.ts:68]: I added numeric separator for readability when I was testing this method

File: api/projects.ts [hunk 2]
```diff
@@ -212,7 +213,7 @@ export async function provisionBuild(
     )}/builds/staged/provision`,
     params: { platformVersion },
     headers: { 'Content-Type': 'application/json' },
-    timeout: 50000,
+    timeout: 50_000,
```
Comment (joe-yeager) [inline api/projects.ts:216]: Same here

File: api/__tests__/projects.test.ts [hunk 3]
```diff
@@ -0,0 +1,577 @@
+jest.mock('../../http');
```
Comment (brandenrodgers) [inline api/__tests__/projects.test.ts:1]: Thanks for adding this! 🙏

**Review by brandenrodgers — APPROVED** [review]
lgtm!

**Pull request merged**